packages/flutter/lib/src/material/dialog.dart

```diff
@@ -117,22 +168,104 @@ class Dialog extends StatelessWidget {
       ));
     }
 
-    return new Center(
-      child: new Container(
-        margin: new EdgeInsets.symmetric(horizontal: 40.0, vertical: 24.0),
+    return new Dialog(
+      child: new IntrinsicWidth(
+        child: new Column(
+          mainAxisSize: MainAxisSize.min,
+          crossAxisAlignment: CrossAxisAlignment.stretch,
+          children: children
```

Comment:
I looked at the callers and they either have something small (e.g., a single text widget with a short string) or they introduce their own scrolling.  I think it's reasonable that we don't introduce scrolling at this level.  Previously, it was confusing because there was two level of possible scrolling.